src/org/mirah/jvm/mirrors/mirror_type_system.mirah

```diff
@@ -115,14 +116,16 @@ class MirrorTypeSystem implements TypeSystem, ExtensionsService
   end
 
   def parameterize(type:TypeFuture, args:List)
-    last_resolved = nil
+    parameterize type, args, {}
+  end
+
+  def parameterize(type:TypeFuture, args:List, processed_signatures:Map)
```

Comment:
I think you should be able to use an optional argument here instead of adding a new declaration.